### PR TITLE
CLOUDP-82909: Allow to append privileges if the privilege already exists

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

_Jira ticket:_ CLOUDP-82909

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/customdbroles/update.go [hunk 1]
```diff
@@ -63,30 +63,41 @@ func (opts *UpdateOpts) Run() error {
...
+	actionMap := make(map[string]atlas.Action)
+	for _, action := range existingActions {
+		actionMap[action.Action] = action
+	}
+	for i, action := range newActions {
+		if a, ok := actionMap[action.Action]; ok {
+			action.Resources = append(action.Resources, a.Resources...)
+			newActions[i] = action
+			delete(actionMap, action.Action)
+		}
```
Comment (andreaangiolillo) [inline internal/cli/atlas/customdbroles/update.go:91]: [q] what is going to happen if I have more than 2 actions that are the same? `FIND@db1.collection1,FIND@db2.collection2,FIND@db3.collection3`

File: internal/cli/atlas/customdbroles/update.go [hunk 2]
```diff
@@ -63,30 +63,41 @@ func (opts *UpdateOpts) Run() error {
...
 
+func appendActions(existingActions, newActions []atlas.Action) []atlas.Action {
+	actionMap := make(map[string]atlas.Action)
+	for _, action := range existingActions {
+		actionMap[action.Action] = action
+	}
+	for i, action := range newActions {
+		if a, ok := actionMap[action.Action]; ok {
+			action.Resources = append(action.Resources, a.Resources...)
+			newActions[i] = action
```
Comment (andreaangiolillo) [inline internal/cli/atlas/customdbroles/update.go:89]: [not bloking] I personally don't like adding/removing/updating an element while iterating over a list.

File: internal/cli/atlas/customdbroles/create.go [hunk 3]
```diff
@@ -28,7 +28,7 @@ import (
 	atlas "go.mongodb.org/atlas/mongodbatlas"
 )
 
-const createTemplate = "Custom Database Role {{.RoleName}} successfully created.\n"
+const createTemplate = "Custom database role '{{.RoleName}}' successfully created.\n"
```
Comment (andreaangiolillo) [inline internal/cli/atlas/customdbroles/create.go:31]: [q] Why can't we provide this functionality to the create command?

File: internal/cli/atlas/customdbroles/update_test.go [hunk 4]
```diff
@@ -73,3 +73,149 @@ func TestUpdateBuilder(t *testing.T) {
...
+						},
+						{
+							Collection: "test",
+							Db:         "test2",
+						},
+					},
+				},
+			},
+		},
+	}
```
Comment (andreaangiolillo) [inline internal/cli/atlas/customdbroles/update_test.go:212]: [q] Can we provide a scenario where we have more than 2 actions that are the same? `FIND@db1.collection1,FIND@db2.collection2,FIND@db3.collection3`

**Review by andreaangiolillo — COMMENTED** [review]
Left some questions

File: internal/cli/atlas/customdbroles/update.go (hunk 1 above)
Comment (gssbzn) [reply internal/cli/atlas/customdbroles/update.go:91]: good catch but that's a different bug of the input so I'll addressed in a different way

File: internal/cli/atlas/customdbroles/create.go (hunk 3 above)
Comment (gssbzn) [reply internal/cli/atlas/customdbroles/create.go:31]: what do you mean here?

Comment (andreaangiolillo) [reply internal/cli/atlas/customdbroles/create.go:31]: I am just wondering why we are not allowing the user to create a custom role with an action which is applied to multiple resources with the `create `command

something like :
`./bin/mongocli atlas customDBRoles create myrole5 --privilege FIND@db1.coll2,FIND@db1.ss,UPDATE@db2.coll3
Error: POST https://cloud-dev.mongodb.com/api/atlas/v1.0/groups/5e4e593f70dfbf1010295836/customDBRoles/roles: 400 (request "ATLAS_DUPLICATE_CUSTOM_ROLE_ACTION") A custom role cannot have duplicate actions.`

Comment (andreaangiolillo) [reply internal/cli/atlas/customdbroles/create.go:31]: but I see now that the `joinActions` should allow this

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Thanks for the changes

File: internal/cli/atlas/customdbroles/create.go (hunk 3 above)
Comment (gssbzn) [reply internal/cli/atlas/customdbroles/create.go:31]: Oh I thought for this comment it was about the line and I was confused

**Pull request merged**